client/clientfactory.go

```diff
@@ -60,6 +59,18 @@ type (
 		NewAdminClientWithTimeout(rpcAddress string, timeout time.Duration, largeTimeout time.Duration) (adminservice.AdminServiceClient, error)
 	}
 
+	// FactoryProvider can be used to provide a customized client Factory implementation.
+	FactoryProvider interface {
+		NewFactory(
+			rpcFactory common.RPCFactory,
+			monitor membership.Monitor,
+			metricsClient metrics.Client,
+			dc *dynamicconfig.Collection,
```

Comment:
should this be an interface?

Reply:
Yes, so this would make it possible to supply customized factory.